src/crypto/algorithms/base.ts

```diff
@@ -66,18 +66,16 @@ export abstract class EncryptionAlgorithm {
     protected readonly crypto: Crypto;
     protected readonly olmDevice: OlmDevice;
     protected readonly baseApis: MatrixClient;
-    protected readonly roomId?: string;
 
     /**
      * @param params - parameters
      */
-    public constructor(params: IParams) {
+    public constructor(params: Omit<IParams, "roomId">) {
```

Comment:
feels like we should leave `params` alone even if we're removing the property on the object, since this defines the common interface for encryptors. (Likewise for `DecryptionAlgorithm`).

This code is all going away soon though, so \<shrug>